docs/auto_arpa.md

```diff
@@ -0,0 +1,91 @@
+## Automatic PTR Generation With auto_arpa
+
+octoDNS supports the automatic generation of `PTR` records for in-addr.arpa. and ip6.arpa. zones. In order to enable the functionality the `auto_arpa` key needs to be passed to the manager configuration.
+
+```yaml
+---
+manager:
+  auto_arpa: true
+```
+
+Alternatively the value can be a dictionary with configuration options for the AutoArpa processor/provider.
+
+```yaml
+---
+manager:
+  auto_arpa:
+    ttl: 1800
+```
+
+Once enabled a singleton `AutoArpa` instance, `auto-arpa`, will be added to the pool of providers and globally configured to run as the very last global processor so that it will see all records as they will be seen by targets. Further all zones ending with `arpa.` will be held back and processed after all other zones have been completed so that all `A` and `AAAA` records will have been seen prior to planning the `arpa.` zones. 
+
+In order to add `PTR` records for a zone the `auto-arpa` source should be added to the list of sources for the zone.
+
+```yaml
+0.0.10.in-addr.arpa.:
+  sources:
+    - auto-arpa
+  targets:
+    - ...
+```
+
+The above will add `PTR` records for any `A` records previously seen with IP addresses 10.0.0.*.
+
+### A Complete Example
+
+#### config/octodns.yaml
+
+```yaml
+manager:
+  auto_arpa: true
+
+providers:
+  config:
+    class: octodns.provider.yaml.YamlProvider
+    directory: tests/config
+
+  powerdns:
+    class: octodns_powerdns.PowerDnsProvider
+    host: 10.0.0.53
+    port: 8081
+    api_key: env/POWERDNS_API_KEY
+
+zones:
+  exxampled.com.:
+    sources:
+      - config
+    targets:
+      - powerdns
+
+  0.0.10.in-addr.arpa.:
```

Comment:
The `Notes` section down below, https://github.com/octodns/octodns/pull/974/files#diff-6a46b87ddab4067f49b72e6939d755b6c22d2e0634499cf0ca3efe2227fa97ceR91, has some info about what happens during partial syncs, but the TL;DR is that ARPA zones are never populated during any sort of partial sync b/c octoDNS can't know what zones are required to get a complete picture of the ARPA zone without planning 100% of things.

You can do partial runs of non-ARPA zones normally, but to auto-manage ARPA zones everything has to be run in one go. This would have prevented us from using this functionality at GitHub as we split things up into multiple runners as well. 

I don't really have great ideas for how to address this. The only thing I can think of to make it possible, with GREAT CARE, is a way to override and ignore the error that gets thrown if there's an ARPA zone during a partial sync. It would take a pretty detailed mapping out of everything on the part of the team that owns the octoDNS implementation to make sure that ARPA zones could never be synced without all their "source" data.

Reply:
Probably easier said than done: can we populate all sources regardless of partial syncs and target supplied zones only during the populate phase?

Reply:
Not currently, but that was something that this situation caused me to think about the need for. Plan all, but apply some should be doable, just not currently. That's probably a new issue tracking that specific functionality. 

/cc https://github.com/octodns/octodns/issues/982 for that issue